webapp/src/components/Competency.tsx

```diff
@@ -1,15 +1,32 @@
-import React from 'react';
+import { Button } from '@mui/material';
+import React, { useContext, useState } from 'react';
+import CreateCompetencyForm from './CreateCompetencyForm';
+import SessionContext from './SessionContext';
+
+import { EntityId } from '../types/api';
 
 interface CompetencyProps {
-  label: string;
   description: string;
+  id: EntityId;
+  label: string;
+  onCompetenciesChanged?: (id: EntityId) => void;
+  principal_id: EntityId;
 }
 
-const Competency = ({ label, description }: CompetencyProps) => (
-  <div>
-    <h2>{label}</h2>
-    <p>{description}</p>
-  </div>
-);
+const Competency = ({ description, id, label, onCompetenciesChanged, principal_id }: CompetencyProps) => {
+  const { principalId } = useContext(SessionContext);
+  const [toggleForm, setToggleForm] = useState(false);
+
+  return (
+    <div>
+      <h2>{label}</h2>
+      <p>{description}</p>
+      {principal_id === principalId && <Button onClick={() => setToggleForm(true)}>Edit Competency</Button>}
```

Comment:
If this is a toggle button, shouldn't it set `toggleForm` to `!toggleForm` instead of `true`?

Reply:
Yes, it should.